code/povelikin-rostislav/src/test/java/ru/unn/agile/redblacktree/RedBlackTreeTest.java

```diff
@@ -0,0 +1,230 @@
+package ru.unn.agile.redblacktree;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+
+import static junit.framework.TestCase.*;
+
+public class RedBlackTreeTest {
+    private static RedBlackTree tree;
+    private static Random random = new Random();
+
+    public RedBlackTreeTest() {
+    }
+
+    @Before
+    public void setUp() {
+        tree = new RedBlackTree();
+    }
+
+    @Test
+    public void defaultConstructedIsEmpty() {
+        assertTrue(tree.isEmpty());
+    }
+
+    @Test
+    public void insertionChangesEmptiness() {
+        tree.insert(8);
+
+        assertFalse(tree.isEmpty());
+    }
+
+    @Test
+    public void emptiedRedBlackTreeIsEmpty() {
+        int expected = 42;
+        tree.insert(expected);
+        tree.remove(expected);
+
+        assertTrue(tree.isEmpty());
+    }
+
+    @Test
+    public void emptyTreeSizeIsZero() {
+        assertEquals(0, tree.getSize());
+    }
+
+    @Test
+    public void notEmptyTreeSizeIsNotZero() {
+        tree.insert(42);
+        assertFalse(tree.getSize() == 0);
+    }
+
+    @Test
+    public void insertionChangesSize() {
+        int threshold = 8;
+        int counter = 0;
+        while (counter < threshold) {
+            ++counter;
+            final int newSize = tree.getSize() + 1;
+            tree.insert(counter);
+            assertEquals(tree.getSize(), newSize);
+        }
+    }
+
+    @Test
+    public void canFindInserted() {
+        int expected = 42;
+        tree.insert(expected);
+
+        assertTrue(tree.find(expected));
+    }
+
+    @Test
+    public void insertCornerCaseMax() {
```

Comment:
Why we need tests like this?

Reply:
The test checks that tree works on the maximum and minimum for the data type we are available to use. It's a kind of check of the range we can use for the tree